hw/xbox/xbox.c

```diff
@@ -253,7 +253,35 @@ void nv2a_init(PCIBus *bus, int devfn, MemoryRegion *ram);
 /* PC hardware initialisation */
 static void xbox_init(MachineState *machine)
 {
-    xbox_init_common(machine, default_eeprom, NULL, NULL);
+    char *filename;
+    int fd;
+    int eeprom_size = 256;
+    int rc;
+    uint8_t *eeprom_data;
+
+    const char *eeprom_file = object_property_get_str(qdev_get_machine(),
+                                                      "eeprom", NULL);
+    if ((eeprom_file != NULL) && *eeprom_file) {
+        filename = qemu_find_file(QEMU_FILE_TYPE_BIOS, eeprom_file);
+        assert(filename);
+
+        eeprom_size = get_image_size(filename);
+        assert(eeprom_size == 256);
+
+        eeprom_data = g_malloc(eeprom_size);
+        fd = open(filename, O_BINARY);
+        assert(fd >= 0);
```

Comment:
I know the bootrom code does this exact same thing, but I think (in both cases) we should fail more gracefully here and print an error message saying that the file could not be opened. It's very possible that we don't have permission or something.